### ISSUE TITLE
Is there a bug in locale validation in "ParserBase::SetExpr"?

### ISSUE THREAD

**Issue by boinst** [beltoforion/muparser#21]
In `ParserBase::SetExpr`, we see the code:

```
 // Check locale compatibility
    std::locale loc;
    if (m_pTokenReader->GetArgSep()==std::use_facet<numpunct<char_type> >(loc).decimal_point())
      Error(ecLOCALE);
```

However, `ParserBase` also has static member `s_locale` defined as follows:

```
static std::locale s_locale;  ///< The locale used by the parser
```

We can manipulate `s_locale` with `SetThousandsSep` and `SetDecSep`. However, when we go to validate our settings, we validate a new `std::locale` instead of `s_locale`. Shouldn't the validation code look more like this? (I've substituted "s_locale" for "loc")

```
 // Check locale compatibility
    if (m_pTokenReader->GetArgSep()==std::use_facet<numpunct<char_type> >(s_locale).decimal_point())
      Error(ecLOCALE);
```

**Comment by beltoforion** [comment]
That code must be more than 10 years old. After checking this i tend to agree with you. Doesn't seem to make much sense in that form. I suspect s_local came in long after that test was written. Did you stumble upon this whilst tracking down an issue in your code?

**Comment by boinst** [comment]
Yes, @beltoforion I came upon this while tracking down an issue in our code. A user in french locale reported the error message "Decimal separator is identic (sic) to function argument separator". I figured I could fix that with `SetDecSep` and `SetThousandsSep`, but inspecting the code, it looks like that wouldn't work at all for the reasons outlined above. I apologise for not seeing your reply earlier, I didn't see the notification. I would be pleased to submit a pull request if it would be useful and if you agree there's a bug here.
